### PR TITLE
Fixes for issues #46 and #47

### PR DESCRIPTION
**Pull request by alankstewart**

### CODE REVIEW

**Comment by alankstewart** [comment]
Retrying should now work for methods returning an Option as well as `createRepository`. `GitRepositoryCloner#clone` now just returns a `File` object and throws an exception in the event of a clone failure.

File: src/main/scala/com/atomist/source/git/GitRepositoryCloner.scala [hunk 1]
```diff
@@ -33,11 +35,9 @@ case class GitRepositoryCloner(oAuthToken: String = "", remoteUrl: Option[String
 
       s"git clone$br --depth $depth --single-branch $getUrl/$owner/$repo.git ${repoDir.getPath}" !! outLogger
       sha.foreach(resetToSha(_, repoDir))
-      Some(repoDir)
+      repoDir
     } catch {
-      case e: Exception =>
-        logger.warn(s"Failed to clone repository $owner/$repo", e)
-        None
+      case e: Exception => throw ArtifactSourceException(s"Failed to clone repository $owner/$repo", e)
```
Comment (kipz) [inline src/main/scala/com/atomist/source/git/GitRepositoryCloner.scala]: I may have missed it, but this could do with a test.

**Comment by kipz** [comment]
LGTM with minor comments.

File: src/main/scala/com/atomist/source/git/GitRepositoryCloner.scala (hunk 1 above)
Comment (alankstewart) [inline src/main/scala/com/atomist/source/git/GitRepositoryCloner.scala]: Added test

**Pull request merged**